pandas/core/indexes/datetimes.py

```diff
@@ -106,15 +106,15 @@ def _dt_index_cmp(opname, cls, nat_result=False):
     """
 
     def wrapper(self, other):
-        func = getattr(super(DatetimeIndex, self), opname)
+        binop = getattr(Index, opname)
 
         if isinstance(other, (datetime, compat.string_types)):
             if isinstance(other, datetime):
                 # GH#18435 strings get a pass from tzawareness compat
                 self._assert_tzawareness_compat(other)
 
             other = _to_m8(other, tz=self.tz)
-            result = func(other)
+            result = binop(self, other)
```

Comment:
why this change?

Reply:
Because `DatetimeIndex.__mro__` has 12 entries, so with the standard `super` usage it isn't obvious where the method is defined.